pkg/storagepool/v1alpha1/kubernetes.go

```diff
@@ -0,0 +1,226 @@
+type createFunc func(cs *clientset.Clientset,
+	storegePoolObj *apis.StoragePool) (*apis.StoragePool, error)
+
+// patchFunc is a typed function that abstracts
+// patching StoragePool instances
+type patchFunc func(cs *clientset.Clientset, name string,
+	pt types.PatchType, patchObj []byte) (*apis.StoragePool, error)
+
+// delFn is a typed function that abstracts
+// delete of StoragePool instances
+type delFn func(cs *clientset.Clientset, name string,
+	opts *metav1.DeleteOptions) error
+
+// Kubeclient enables kubernetes API operations
+// on StoragePool instance
+type Kubeclient struct {
+	// clientset refers to StoragePool's
+	// clientset that will be responsible to
+	// make kubernetes API calls
+	clientset *clientset.Clientset
+	// functions useful during mocking
+	getClientset getClientsetFunc
+	list         listFunc
+	get          getFunc
+	create       createFunc
+	patch        patchFunc
+	del          delFn
+}
+
+// KubeclientBuildOption defines the abstraction
+// to build a Kubeclient instance
+type KubeclientBuildOption func(*Kubeclient)
+
+// withDefaults sets the default options
+// of kubeclient instance
+func (k *Kubeclient) withDefaults() {
+	if k.getClientset == nil {
+		k.getClientset = func() (cs *clientset.Clientset, err error) {
+			config, err := client.GetConfig(client.New())
+			if err != nil {
+				return nil, err
+			}
+			return clientset.NewForConfig(config)
+		}
+	}
+
+	if k.list == nil {
+		k.list = func(cs *clientset.Clientset,
+			opts metav1.ListOptions) (*apis.StoragePoolList, error) {
+			return cs.OpenebsV1alpha1().
+				StoragePools().
+				List(opts)
+		}
+	}
+
+	if k.get == nil {
+		k.get = func(cs *clientset.Clientset,
+			name string, opts metav1.GetOptions) (*apis.StoragePool, error) {
+			return cs.OpenebsV1alpha1().
+				StoragePools().
+				Get(name, opts)
+		}
+	}
+
+	if k.create == nil {
+		k.create = func(cs *clientset.Clientset,
+			storagePoolObj *apis.StoragePool) (*apis.StoragePool, error) {
+			return cs.OpenebsV1alpha1().
+				StoragePools().
+				Create(storagePoolObj)
+		}
+	}
+
+	if k.patch == nil {
+		k.patch = func(cs *clientset.Clientset, name string,
+			pt types.PatchType, patchObj []byte) (*apis.StoragePool, error) {
+			return cs.OpenebsV1alpha1().
+				StoragePools().
+				Patch(name, pt, patchObj)
+		}
+	}
+
+	if k.del == nil {
+		k.del = func(cs *clientset.Clientset, name string,
+			opts *metav1.DeleteOptions) error {
+			return cs.OpenebsV1alpha1().
+				StoragePools().
+				Delete(name, opts)
+		}
+	}
+}
+
+// WithClientset sets the kubernetes clientset against
+// the kubeclient instance
+func WithClientset(c *clientset.Clientset) KubeclientBuildOption {
+	return func(k *Kubeclient) {
+		k.clientset = c
+	}
+}
+
+// NewKubeClient returns a new instance of kubeclient meant for
+// StoragePool related operations
+func NewKubeClient(opts ...KubeclientBuildOption) *Kubeclient {
+	k := &Kubeclient{}
+	for _, o := range opts {
+		o(k)
+	}
+	k.withDefaults()
+	return k
+}
+
```

Comment:
can we have WithFlag for storagepool resource.

Reply:
can be done in a different PR.